python/spec/dependabot/python/file_parser_spec.rb

```diff
@@ -215,9 +215,9 @@
         )
       end
 
-      describe "the first dependency" do
-        it "has the right details" do
-          expect(dependencies.map(&:name)).not_to match_array([])
+      describe "the tarball dependency requirement" do
+        it "is not parsed" do
+          expect(dependencies).to eq([])
```

Comment:
> Could you clarify the changes to this test a bit? 

Absolutely!

> How does that end up in not parsing the first (tarball?) 

This change actually brings our tests in line with the current behavior, which was the result of a [Poetry-specific PR](https://github.com/dependabot/dependabot-core/pull/3991). The production-code changes in that PR left us in a state where we only parsed `requests`, so the assertion of `expect(dependencies.map(&:name)).not_to match_array([])` was valid because the array was actually `["requests"]`.

> why did we remove requests from the fixture?

Mostly because of ☝️, which I think was a false positive. The test was meant to be parsing tarball requirements, but was passing because of the non-tarball requirements.

---

Whether this tested behavior is intended, I'm not 100% sure. However, I reasoned that accuracy was more important than intention.